plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/TemplateListPage.java

```diff
@@ -164,21 +195,136 @@ public Object convert(Object fromObject) {
+		dbc.addValidationStatusProvider(pageValidator );
+		ControlDecorationSupport.create(pageValidator, SWT.LEFT | SWT.TOP, null, new RequiredControlDecorationUpdater(true));
+	}
+
+	private IObservableValue createEclipseProjectControls(Composite parent, DataBindingContext dbc) {
+		// existing project
+		Label existingProjectLabel = new Label(parent, SWT.NONE);
+		existingProjectLabel.setText("Use existing workspace project:");
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.CENTER)
+				.grab(false, false)
+				.applyTo(existingProjectLabel);
+
+		final Text existingProjectNameText = new Text(parent, SWT.BORDER);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.CENTER)
+				.grab(true, false)
+				.applyTo(existingProjectNameText);
+		
+		ISWTObservableValue projectNameTextObservable = WidgetProperties.text(SWT.Modify).observe(existingProjectNameText);
+
+		IObservableValue eclipseProjectObservable = BeanProperties.value(
+				ITemplateListPageModel.PROPERTY_ECLIPSE_PROJECT).observe(model);
+		
+		ValueBindingBuilder
+			.bind(projectNameTextObservable)
+			.converting(new Converter(String.class, org.eclipse.core.resources.IProject.class) {
+				@Override
+				public Object convert(Object fromObject) {
+					String name = (String)fromObject;
+					if(StringUtils.isNotBlank(name)) {
+						org.eclipse.core.resources.IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(name);
+						if (ProjectUtils.exists(project)) {
+							return project;
+						}
+					}
+					return null;
+				}
+			})
+			.to(eclipseProjectObservable)
+			.converting(new Converter(org.eclipse.core.resources.IProject.class, String.class) {
+				
+				@Override
+				public Object convert(Object fromObject) {
+					return fromObject == null?"": ((org.eclipse.core.resources.IProject)fromObject).getName();
+				}
+			})
+			.in(dbc);
+
+		// project name content assist
+		ControlDecoration dec = new ControlDecoration(existingProjectNameText, SWT.TOP | SWT.RIGHT);
+		
+		FieldDecoration contentProposalFieldIndicator =
+				FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_CONTENT_PROPOSAL);
+		dec.setImage(contentProposalFieldIndicator.getImage());
+		dec.setDescriptionText("Auto-completion is enabled when you start typing a project name.");
+		dec.setShowOnlyOnFocus(true);
+
+		new AutoCompleteField(existingProjectNameText, new TextContentAdapter(), ProjectUtils.getAllOpenedProjects());
+
+		// browse projects
+		Button browseProjectsButton = new Button(parent, SWT.NONE);
+		browseProjectsButton.setText("Browse...");
+		GridDataFactory.fillDefaults()
+				.align(SWT.LEFT, SWT.CENTER)
+				.hint(100, SWT.DEFAULT)
+				.grab(false, false)
+				.applyTo(browseProjectsButton);
+		browseProjectsButton.addSelectionListener(onBrowseProjects());
+		
+		Link gitLabel = new Link(parent, SWT.NONE);
```

Comment:
Yes. Whether the wizard is opened from the Deploy to OpenShift menu, JBoss Central, or OpenShift Explorer.